### PR TITLE
Broker id must remain the same during kafka instances restart

### PR DESCRIPTION
**Pull request by antban**

### CODE REVIEW

File: src/id_policy.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    elif addr_data[0] == 172 and (addr_data[1] & 0xF0) == 16:
+        addr_data[0] = 3
+    else:
+        return None
+    return str(functools.reduce(lambda o, v: o * 256 + v, addr_data, 0))
+
+
+def _wait_for_zk_node_absence(zk: KazooClient, node: str):
+    try:
+        while zk.get(node):
```
Comment (rcillo) [inline src/id_policy.py]: Is it guaranteed to eventually jump out of this loop?

File: src/id_policy.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                    return match.group(1)
+        return None
+
+
+def get_broker_policy(policy, kafka_dir, kafka_data_dir):
+    if policy == 'ip':
+        return BrokerIDByIp(kafka_dir)
+    elif policy == 'auto':
+        return BrokerIDAutoAssign(kafka_data_dir)
+    return BrokerIDByIp(kafka_dir)
```
Comment (rcillo) [inline src/id_policy.py]: Why not to fail? Otherwise someone could be providing a magic value that in the future would be mistakenly used by newly added elif branch.

File: src/id_policy.py (hunk 1 above)
Comment (antban) [inline src/id_policy.py]: If kafka is not stopping (stopped) it's guaranteed that we won't exit. 
But if it's stopped (stopping), ephemeral node /broker_ids/id will be defenitely removed.

File: src/id_policy.py (hunk 2 above)
Comment (antban) [inline src/id_policy.py]: This logic was here before, I do not want to break it in this commit.

File: src/id_policy.py (hunk 1 above)
Comment (rcillo) [inline src/id_policy.py]: Let me rephrase my question. This code guarantees that a broker won't accidentally generate a Broker ID with an increment of 1. In order to do so, it waits for the node to be gone. My question is: is there the possibility for this Node be used by another broker? I mean, in case of conflict. Then this broker would hang here waiting for that indefinitely.

Comment (antban) [inline src/id_policy.py]: Well, there is a theoretical possibility. If id is generated from private ip address, it's guaranteed that this id is greater or equal to 16777216. I think that it's impossible to create this broker id using +1 technique.
